x-pack/plugins/integration_assistant/server/graphs/log_type_detection/graph.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ActionsClientChatOpenAI,
+  ActionsClientSimpleChatModel,
+} from '@kbn/langchain/server/language_models';
+import type { StateGraphArgs } from '@langchain/langgraph';
+import { END, START, StateGraph } from '@langchain/langgraph';
+import type { LogFormatDetectionState } from '../../types';
+import { EX_ANSWER_LOG_TYPE } from './constants';
+import { handleLogFormatDetection } from './detection';
+import { SamplesFormat } from '../../../common';
+
+const graphState: StateGraphArgs<LogFormatDetectionState>['channels'] = {
+  lastExecutedChain: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  rawSamples: {
+    value: (x: string[], y?: string[]) => y ?? x,
+    default: () => [],
+  },
+  exAnswer: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  finalized: {
+    value: (x: boolean, y?: boolean) => y ?? x,
+    default: () => false,
+  },
+  samplesFormat: {
+    value: (x: SamplesFormat, y?: SamplesFormat) => y ?? x,
+    default: () => ({ name: 'json' }),
+  },
+  ecsVersion: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '8.11.0',
+  },
+  results: {
+    value: (x: object, y?: object) => y ?? x,
+    default: () => ({}),
+  },
+};
+
+function modelInput(state: LogFormatDetectionState): Partial<LogFormatDetectionState> {
+  return {
+    exAnswer: JSON.stringify(EX_ANSWER_LOG_TYPE, null, 2),
+    finalized: false,
+    lastExecutedChain: 'modelInput',
+  };
+}
+
+function modelOutput(state: LogFormatDetectionState): Partial<LogFormatDetectionState> {
+  return {
+    finalized: true,
+    lastExecutedChain: 'modelOutput',
+    results: {
+      samplesFormat: state.samplesFormat,
+      parsedSamples: state.rawSamples, // TODO: Add parsed samples
+    },
+  };
+}
+
+function logFormatRouter(state: LogFormatDetectionState): string {
+  // if (state.logFormat === LogFormat.STRUCTURED) {
+  //   return 'structured';
+  // }
```

Comment:
Should we maybe make the graph return the correct type already? And just have either the backend check if its anything but json? Or even the UI I guess we could test it?

Reply:
The graph returns correctly. That is handled in `detection.js`. These are more for the Router in the graph to route to sub graphs based on the log format.

Like KV graph for a structured log format , which is not implemented yet.

Reply:
The router checks for this [here](https://github.com/elastic/kibana/pull/190407/files#diff-3db1d6aa72318e024ba8ad4874bf126fd68ca01138aba3e8be7bb6e6a62cd62cR99-R102)